src/dc-factory.ts

```diff
@@ -0,0 +1,164 @@
+        rComponent.components.find(({ element }) => element === _element);
+
+    private _getState = (element: HTMLElement, rComponent: RegisteredComponent): string => {
+        const component = this._getComponent(element, rComponent);
+        if (component) return component.state;
+        return STATE_NOT_INITED;
+    }
+
+    private _setState = (element: HTMLElement, rComponent: RegisteredComponent, state: string, instance?: DcBaseComponent | undefined): void => {
+        let component = this._getComponent(element, rComponent);
+
+        if (!component) {
+            component = { element, state, instance }
+            rComponent.components.push(component);
+        }
+        Object.assign(component, { state, instance })
+    }
+
+    private _initComponent = (root: HTMLElement, rComponent: RegisteredComponent, withLazy?: boolean): void => {
+        try {
+            const { _Class: { namespace }, selector } = rComponent;
+            findElementsForInit(root, namespace, selector).forEach((element) => {
+                this._initComponentOnElement(element, rComponent, withLazy);
+            });
+        } catch (e) {
+            // ignore current config error and move to the next one
+            console.error(e);
+        }
+    }
+
+    private _initComponentOnElement = (element: HTMLElement, rComponent: RegisteredComponent, withLazy?: boolean): void => {
+        const { _Class } = rComponent;
+        const state = this._getState(element, rComponent);
+        switch (state) {
+            // ignore components which are already created or in the middle of that process
+            case STATE_CREATED:
+            case STATE_ERROR:
+            case STATE_INITIALIZING:
+                return;
+            case STATE_LAZY_WAITING:
+                if (!withLazy) {
+                    return;
+                }
+        }
+
+        // if component is lazy but we should not instantiate it according withLazy = false
+        // we need to mark this component and wait until withLazy = true
+        if (!withLazy && isElementWithinLazyParent(element)) {
+            this._setState(element, rComponent, STATE_LAZY_WAITING);
+            return;
+        }
+
+        // finally init component on element
+        this._setState(element, rComponent, STATE_INITIALIZING);
+        requestAnimationFrame(() => {
+            try {
+                this._createComponent(element, rComponent);
+            } catch (error) {
+                this._setState(element, rComponent, STATE_ERROR);
+                console.error(`Component ${_Class.name} hasn't been created due to error: ${error.message}`, element);
+                console.error(error);
+            }
+        });
+    }
+
+
+    private _createComponent = (element: HTMLElement, rComponent: RegisteredComponent): void => {
+        const { _Class } = rComponent;
+        const instance = new _Class(element, _Class.namespace, _Class.requiredRefs);
+        console.log(instance);
```

Comment:
Should we get rid off of the console statement here?

Reply:
oh yes, I forgot to remove it after debugging